inference-engine/src/vpu/common/src/ngraph/operations/dynamic_shape_resolver.cpp

```diff
@@ -30,18 +34,159 @@ void DynamicShapeResolver::validate_and_infer_types() {
                                                                 dimsElementType.compatible(ngraph::element::i32)),
         "(", get_friendly_name(), ") supports only i64 and i32 number type for dims tensor, but ", dimsElementType, " provided");
 
-    const auto& dataShape = get_input_shape(0);
     const auto& dimsShape = get_input_shape(1);
-    NODE_VALIDATION_CHECK(this, dimsShape.size() == 1 && dimsShape.front() == dataShape.size(), "(", get_friendly_name(), ") inputs shapes mismatch: first "
-        "input shape = ", dataShape, " second input shape = ", dimsShape, " but ", dataShape, " and ", Shape{dataShape.size()}, " are expected");
 
-    set_output_type(0, dataElementType, dataShape);
+    if (m_mode == DynamicShapeResolverMode::INFER_UPPER_BOUND_SHAPE) {
+        NODE_VALIDATION_CHECK(this, get_input_partial_shape(0).is_static(), "(", get_friendly_name(), ") does not support dynamic shape for data tensor");
+
+        const auto& dataShape = get_input_shape(0);
+        NODE_VALIDATION_CHECK(this, dimsShape.size() == 1 && dimsShape.front() == dataShape.size(), "(", get_friendly_name(), ") inputs shapes mismatch: first "
+            "input shape = ", dataShape, " second input shape = ", dimsShape, " but ", dataShape, " and ", Shape{dataShape.size()}, " are expected");
+
+        set_output_type(0, dataElementType, dataShape);
+    } else {
+        NODE_VALIDATION_CHECK(this, get_input_partial_shape(0).rank() == dimsShape.front(),
+                "(", get_friendly_name(), ") data and shape ranks must be equal, proided: ",
+                get_input_partial_shape(0).rank(), " vs ", dimsShape.front());
+
+        set_output_type(0, dataElementType,
+                        ngraph::PartialShape::dynamic(get_input_partial_shape(0).rank()));
+    }
 }
 
 bool DynamicShapeResolver::visit_attributes(ngraph::AttributeVisitor&) {
     return true;
 }
 
+namespace {
+
+template<element::Type_t ET>
+bool getShapeFromHostTensorData(const HostTensorPtr& data, Shape& result) {
+    using T = typename element_type_traits<ET>::value_type;
+    T *dataPtr = data->get_data_ptr<ET>();
+    if (!dataPtr) {
+        return false;
+    }
+    size_t outputRank = data->get_shape()[0];
```

Comment:
Do we need to check there that there is only 1 element?

Reply:
Fixed